aspnetcore/security/authentication/certauth.md

```diff
@@ -206,3 +206,249 @@ Complete the following steps In IIS Manager:
 ### Azure and custom web proxies
 
 See the [host and deploy documentation](xref:host-and-deploy/proxy-load-balancer#certificate-forwarding) for how to configure the certificate forwarding middleware.
+
+### Using certificate authentication with the X-ARR-ClientCert header
+
+The AddCertificateForwarding method is used so that the client header can be specified and how the certificate is to be loaded using the HeaderConverter option. When sending the certificate with the HttpClient using the default settings, the ClientCertificate was always be null. The X-ARR-ClientCert header is used to pass the client certificate, and the cert is passed as a string to work around this.
+
+```csharp
+services.AddCertificateForwarding(options =>
+{
+	options.CertificateHeader = "X-ARR-ClientCert";
+	options.HeaderConverter = (headerValue) =>
+	{
+		X509Certificate2 clientCertificate = null;
+		if(!string.IsNullOrWhiteSpace(headerValue))
+		{
+			byte[] bytes = StringToByteArray(headerValue);
+			clientCertificate = new X509Certificate2(bytes);
+		}
+
+		return clientCertificate;
+	};
+});
+```
+
+The Configure method then adds the middleware. UseCertificateForwarding is added before the UseAuthentication and the UseAuthorization.
+
+```csharp
+public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+{
+	...
+	
+	app.UseRouting();
+
+	app.UseCertificateForwarding();
+	app.UseAuthentication();
+	app.UseAuthorization();
+
+	app.UseEndpoints(endpoints =>
+	{
+		endpoints.MapControllers();
+	});
+}
+```
+
+A separate class can be used to implement validation logic. Because we are using the same self signed certificate in this example, we need to ensure that only our certificate can be used. We validate that the thumbprints of the client certificate and also the server one match, otherwise any certificate can be used and will be be enough to authenticate. This would be used inside the AddCertificate method. You could also validate the subject or the issuer here, if you are using intermediate or child certificates.
```

Comment:
```suggestion
A separate class can be used to implement validation logic. Because the same self-signed certificate is used in this example, ensure that only your certificate can be used. Validate that the thumbprints of both the client certificate and the server certificate match, otherwise any certificate can be used and will be enough to authenticate. This would be used inside the `AddCertificate` method. You could also validate the subject or the issuer here if you're using intermediate or child certificates.

```

Reply:
done